tools/psoxy-test/lib/utils.js

```diff
@@ -64,8 +65,9 @@ function getFileNameFromURL(url, timestamp = true, extension = '.json') {
  */
 function getCommonHTTPHeaders(options = {}) {
   const headers = {
-    'Accept-Encoding': options.gzip ? 'gzip' : '*',
+    'Accept-Encoding': options.gzip ? 'gzip,deflate' : '*',
     'X-Psoxy-Skip-Sanitizer': options.skip?.toString() || 'false',
+    'User-Agent': 'psoxy-test (gzip)', // w/o (gzip) here, GCP Cloud functions don't compress response
```

Comment:
weird, how did you discover this? and also, is this happening to regular requests from GAE??? are we sure we are compressing the responses in GCP?

Reply:
yes, i was testing everything as had a cost concern from a customer, and reducing network would be quick win - but it's OK.  

GCP handles compression auto-magically for cloud functions as long as the `Accepts` and `User-Agent` follow what they expect; on the Worklytics-side we already have something in the client factory that's adding an `HttpRequestInitializer` which adds these headers for proxy connections to GCP.

Separately, I've added code on the proxy-side to add warnings to both logs and response headers if compression is NOT requested by the client.